chimney/src/main/scala/io/scalaland/chimney/dsl.scala

```diff
@@ -32,6 +32,11 @@ object dsl {
     ): TransformerInto[From, To, Modifier.relabel[labelFrom.T, labelTo.T] :: Modifiers] =
       new TransformerInto(source, new Modifier.relabel[labelFrom.T, labelTo.T] :: modifiers)
 
+    def withCoproductInstance[Inst](
+      f: Inst => To
+    ): TransformerInto[From, To, Modifier.coproductInstance[Inst, To] :: Modifiers] =
+      new TransformerInto(source, new Modifier.coproductInstance[Inst, To](f) :: modifiers)
+
     def transform(implicit transformer: DerivedTransformer[From, To, Modifiers]): To =
```

Comment:
Inside internals order should not matter as much... but it would "feel" better for future users that might want to implement `implicit val t: Transformer[...] = ...` when whatever they are impementing matches the signature of a `.transform` call - otherwise we have a situation when user creates `Transformer` and method accepts `DerivedTransformer`. I think that potentially that might cause a trouble to users. But that's for the future to discuss.